### PR TITLE
`azurerm_bot_channels_registration` - remove `Computed` from `developer_app_insights_api_key`, `developer_app_insights_application_id`, `developer_app_insights_key` and `icon_url`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description
This PR is to remove `Computed` from `developer_app_insights_api_key`, `developer_app_insights_application_id`, `developer_app_insights_key` and `icon_url`.

For `developer_app_insights_api_key`, `developer_app_insights_application_id` and `developer_app_insights_key`, we can directly remove `Computed` since service API wouldn't return value when they aren't set.

For `icon_url`, I added default value since service API returns default value when it isn't set.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [x] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

The failed test case is also failed with same error on Teamcity Daily Run. So it's not related with this PR.
image

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_bot_channels_registration` - remove `Computed` from `developer_app_insights_api_key`, `developer_app_insights_application_id`, `developer_app_insights_key` and `icon_url`

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: website/docs/r/bot_channels_registration.html.markdown [hunk 1]
```diff
@@ -65,6 +65,8 @@ The following arguments are supported:
 
 * `icon_url` - (Optional) The icon URL to visually identify the Bot Channels Registration.
 
+-> **NOTE:** In v4.0 of the provider the `icon_url` will have a default value of `https://docs.botframework.com/static/devportal/client/images/bot-framework-default.png`.
```
Comment (stephybun) [inline website/docs/r/bot_channels_registration.html.markdown:68]: Let's omit the 4.0 note in the docs, we will add a full list of properties that have been affect in the upgrade guide closer to the release 
```suggestion
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @neil-yechenwei. Would you mind looking at the two comments I left in-line? Once they're resolved this should be good to go.

File: internal/services/bot/bot_channels_registration_resource_test.go [hunk 2]
```diff
@@ -282,6 +282,7 @@ resource "azurerm_bot_channels_registration" "test" {
   endpoint                              = "https://example2.com"
   developer_app_insights_api_key        = azurerm_application_insights_api_key.test2.api_key
   developer_app_insights_application_id = azurerm_application_insights.test2.app_id
+  developer_app_insights_key            = azurerm_application_insights.test2.instrumentation_key
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply internal/services/bot/bot_channels_registration_resource_test.go:285]: Currently test case always don't set this property. So I think we should update the test scenario to test the case like updating developer_app_insights_key from a value to empty.

File: website/docs/r/bot_channels_registration.html.markdown (hunk 1 above)
Comment (neil-yechenwei) [reply website/docs/r/bot_channels_registration.html.markdown:68]: Removed it.

**Comment by neil-yechenwei** [comment]
@stephybun , thanks for the comments. I updated PR. Please take another look. Thanks.

**Review by stephybun — APPROVED** [review]
Thanks @neil-yechenwei LGTM 👍

**Pull request merged**